clients/py/sapphirepy/sapphire.py

```diff
@@ -63,79 +87,152 @@ def _encrypt_tx_params(pk:CalldataPublicKey, params:tuple[TxParams]):
     else:
         raise TypeError("Invalid 'data' type", type(data))
     encrypted_data = c.encrypt(data_bytes)
+
+    # if False:
+    if params[0]['from'] and params[0]['from'] == account.address:
+        domain_data = {
+            "name": "oasis-runtime-sdk/evm: signed query",
+            "version": "1.0.0",
+            "chainId": web3.eth.chain_id,
+            # "verifyingContract": "",
+            # "salt": "",
+        }
+        msg_types = {
+            "Call": [
+                {"name": "from", "type": "address"},
+                {"name": "to", "type": "address"},
+                {"name": "gasLimit", "type": "uint64"},
+                {"name": "gasPrice", "type": "uint256"},
+                {"name": "value", "type": "uint256"},
+                {"name": "data", "type": "bytes"},
+                {"name": "leash", "type": "Leash"},
+            ],
+            "Leash": [
+                {"name": "nonce", "type": "uint64"},
+                {"name": "blockNumber", "type": "uint64"},
+                {"name": "blockHash", "type": "bytes32"},
+                {"name": "blockRange", "type": "uint64"},
+            ],
+        }
+        msg_data = {
+            "from": params[0].get('from'),
+            "to": params[0].get('to', '0x'),
+            "value": params[0].get('value', 0),
+            "gasLimit": params[0].get('gas', DEFAULT_GAS_LIMIT),
+            "gasPrice": params[0].get('gasPrice', DEFAULT_GAS_PRICE),
+            "data": params[0].get('data'),
+            "leash":
+                {
+                    "nonce": web3.eth.get_transaction_count(params[0].get('from')),
```

Comment:
Extract block and nonce into variable, to prevent multiple RPC requests for same data.